pkg/service/blobindexlookup/simplelookup.go

```diff
@@ -25,27 +31,98 @@ func NewBlobIndexLookup(httpClient *http.Client) BlobIndexLookup {
 }
 
 // Find fetches the blob index from the given fetchURL
-func (s *simpleLookup) Find(ctx context.Context, _ types.EncodedContextID, _ model.ProviderResult, fetchURL *url.URL, rng *metadata.Range) (blobindex.ShardedDagIndexView, error) {
+func (s *simpleLookup) Find(ctx context.Context, _ types.EncodedContextID, result model.ProviderResult, request types.RetrievalRequest) (blobindex.ShardedDagIndexView, error) {
+	// If retrieval authroization details were provided, make a UCAN authorized
+	// retrieval request.
+	if request.Auth != nil {
+		body, err := doAuthorizedRetrieval(ctx, s.httpClient, request)
+		if err != nil {
+			return nil, fmt.Errorf("executing authorized retrieval: %w", err)
+		}
+		defer body.Close()
+		return blobindex.Extract(body)
+	}
+
 	// attempt to fetch the index from provided url
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, fetchURL.String(), nil)
+	httpReq, err := http.NewRequestWithContext(ctx, http.MethodGet, request.URL.String(), nil)
```

Comment:
I think encapsulating the non-auth retrieval branch in a `doRegularRetrieval` function would make the method more readable.